### PR TITLE
feat: Append previous workflow name as label to resubmitted workflow

### PR DESCRIPTION
**Pull request by terrytangyuan**
Append an additional label so it's easy for user to see the name of the original workflow that has been resubmitted.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed the CLA.
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

### CODE REVIEW

File: workflow/util/util.go [hunk 1]
```diff
@@ -444,7 +444,10 @@ func updateWorkflowNodeByKey(wfIf v1alpha1.WorkflowInterface, hydrator hydrator.
 	return err
 }
 
-const letters = "abcdefghijklmnopqrstuvwxyz0123456789"
+const (
+	letters                      = "abcdefghijklmnopqrstuvwxyz0123456789"
+	previousWorkflowNameLabelKey = "workflows.argoproj.io/previous-workflow-name"
```
Comment (simster7) [inline workflow/util/util.go:449]: This should be added in https://github.com/argoproj/argo/blob/master/workflow/common/common.go

Also I would suggest this be called
```suggestion
	previousWorkflowNameLabelKey = "workflows.argoproj.io/resubmitted-from-workflow"
```

File: workflow/util/util.go [hunk 2]
```diff
@@ -493,15 +496,18 @@ func FormulateResubmitWorkflow(wf *wfv1.Workflow, memoized bool) (*wfv1.Workflow
 
 	// carry over user labels and annotations from previous workflow.
 	// skip any argoproj.io labels except for the controller instanceID label.
+	if newWF.ObjectMeta.Labels == nil {
+		newWF.ObjectMeta.Labels = make(map[string]string)
+	}
```
Comment (simster7) [inline workflow/util/util.go:501]: Huh, wonder why this didn't come up as an issue before

File: workflow/util/util.go [hunk 3]
```diff
@@ -493,15 +496,18 @@ func FormulateResubmitWorkflow(wf *wfv1.Workflow, memoized bool) (*wfv1.Workflow
...
 			continue
 		}
-		if newWF.ObjectMeta.Labels == nil {
-			newWF.ObjectMeta.Labels = make(map[string]string)
-		}
 		newWF.ObjectMeta.Labels[key] = val
 	}
+	// Append an additional label so it's easy for user to see the
+	// name of the original workflow that has been resubmitted.
+	newWF.ObjectMeta.Labels[previousWorkflowNameLabelKey] = wf.ObjectMeta.Name
```
Comment (simster7) [inline workflow/util/util.go:510]: 👍

**Comment by terrytangyuan** [comment]
@simster7 Thanks for the review! Just updated the PR to address your comments.

**Pull request merged**